### PR TITLE
Reset start/stop needle sides when loading an image

### LINKED ISSUE #732: [BUG] Opening a new image does not reset the Left and Right settings
AYAB software version: 1.0Beta3
Computer/OS: Mac
Knitting machine: 930
AYAB hardware: shield

If you set the start and stop edges to both be to the left of center, then open a new image, it doesn't switch the stop edge back to the right.

1. Open any image, say the Triangles test image. All settings on default. It correctly defaults to needles L30-R30
2. Change the Start and stop settings to Left90 - Left 30. 
3. Load a new image, can be the same one again
Expected: it should reset the settings to Left and Right with the image centered
Actual: It sets the new needle numbers correctly, but both settings are still on the left, it has not reset the Left/Right setting.

Related to the unfinished discussion in #599 about what it should or shouldn't remember or reset from one image to the next, but this behavior is just wrong.

### PR DESCRIPTION
**Pull request by jonathanperret**
## Problem

Fixes #732.

## Proposed solution

Reset start/stop needle sides when loading an image.

## How to test

Try this test release: https://github.com/jonathanperret/ayab-desktop/releases/tag/1.0.0-startstop-side-reset-1

## Summary by CodeRabbit

- **New Features**
	- Enhanced the `OptionsTab` to automatically select needle colors when image dimensions are set.
  
- **Bug Fixes**
	- Improved validation logic for `portname` and needle color selection to prevent errors.

- **Improvements**
	- Added restrictions on the number of colors based on selected modes, ensuring clarity in user options.

### CODE REVIEW

**Review by dl1com — APPROVED** [review]

**Comment by jonathanperret** [comment]
~Thanks @dl1com, now waiting for @Adrienne200 's feedback before merging.~

I see you've merged this already 😅

**Pull request merged**
